services/efs-guardian/src/simcore_service_efs_guardian/rpc/rpc_routes.py

```diff
@@ -0,0 +1,22 @@
+from collections.abc import Awaitable, Callable
+
+from fastapi import FastAPI
+
+
+def on_app_startup(app: FastAPI) -> Callable[[], Awaitable[None]]:
+    async def _start() -> None:
+        assert app  # nosec
+
+    return _start
+
+
+def on_app_shutdown(app: FastAPI) -> Callable[[], Awaitable[None]]:
+    async def _stop() -> None:
+        assert app  # nosec
+
+    return _stop
+
+
+def setup_rpc_routes(app: FastAPI) -> None:
+    app.add_event_handler("startup", on_app_startup(app))
```

Comment:
we need to create an example that uses lifespan events (https://fastapi.tiangolo.com/advanced/events/) which is the new way ... the more we wait the harder is going to be the refactoring

Reply:
yes, that valid point, I have added it to my TODO list.

Reply:
it does not work for multiple setups @pcrespov I already checked last time. unless they changed it